c++/include/orc/Vector.hh

```diff
@@ -248,10 +254,16 @@ namespace orc {
     ~EncodedStringVectorBatch() override;
     std::string toString() const override;
     void resize(uint64_t capacity) override;
+
+    // Calculate data and length in StringVectorBatch from dictionary and index
+    void decodeDictionary() override;
+
     std::shared_ptr<StringDictionary> dictionary;
 
     // index for dictionary entry
     DataBuffer<int64_t> index;
+
+    std::atomic<bool> dictionaryDecoded{false};
```

Comment:
I think it is an overkill to use atomic here. It should be a misuse to call member functions of ColumnVectorBatch concurrently.

Reply:
It's a personal coding habit in case there might be a multi-threaded call scenario in the future. Besides, `std::atomic<bool>` is so trivial that it would not influence the performance.

Reply:
Anyway, I'll remove it to keep code simple.